### PR TITLE
QC Rework Quest Changes

### PR DESCRIPTION
**Pull request by DylanTaylor1**
**THIS IS HEAVILY DEPENDENT ON THE QC REWORK https://github.com/GTNewHorizons/GT5-Unofficial/pull/2849**

The previous QC quest and "Play it Again, Sam" were both really unhelpful in understanding the QC and did not provide any numbers, equations, or really any information about the machine. They really just told the player to "do this" without much explanation as to why.

These new quests closely resemble the wiki page and provide a significant amount of information in an organized and readable manner. The goal is to give the player enough of an understanding that they can begin working with the QC without having to go to the wiki right away.

### CODE REVIEW

**Comment by Pxx500** [comment]
Small nitpick
image
changing this to "e.g. [insert one or two multies]" will futureproof the quest so it won't need updating when a new multi using computation gets added.

Generally putting numbers into quests makes them outdated when any changes happen, it would generally be better to say something like "[x type of item] is eligible to generate computation, look in NEI tooltips to find the numbers by searching [xyz]" (new nei supports tooltip search) and actually adding info about those numbers into the tooltip, this makes maintaining questbook easier (though I understand just shoving numbers into the quest is easier so I'm not actually opposed to this PR)

Other than that it looks good 👍

**Comment by DylanTaylor1** [comment]
Thanks! And there could certainly be a whole debate on where to put the numbers. Previously they were not available anywhere besides the code and the wiki page which worked fine especially if there was just ONE best setup that everyone talked about. 

Putting them in the tooltip of the circuits might be a little confusing/cluttered because this is a very niche use of the circuits. Putting them in a quest is not super future proof like you mentioned, but at least it's somewhere and easy to edit. It could go many different ways but I think so long as there is one way to find them in-game and one way to find them out-of-game (wiki) is good.

**Comment by chochem** [comment]
I would say tables of numbers fit better on a wiki page or spreadsheet than a quest.

maybe a custom NEI page would be a really nice in-game possiblity. But such a table in a quest text is pretty awkward

File: config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumComputer-AAAAAAAAAAAAAAAAAAAH1w==.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
...
... middle and two sets of 8 buttons on the left and right. Clicking on one button and then another will switch the states of the two corresponding cells in the matrix. This means switching their blinking speed (in the basic Uncertainty Resolver) or their shade/color (in the Uncertainty Resolver X). The goal of the puzzle is to balance the states of the matrix around the glowing LED lights, turning all of them green.\n\n§lPARAMETERS§r. In the GUI of the controller, there are two configurable parameters, overclock and overvolt, that change how computation and heat are generated. Higher overclock values contribute the most to computation and higher overvolt values contribute the most to heat. Alone, this would encourage the player to make overclock as high as possible and overvolt as low as possible, but there is a penalty to computation for large differences between these two parameters. Furthermore, overvolt values \u003c1 will cause the QC to randomly drop computation proportional to its distance from 1. It is highly encouraged to play around with these parameters, as all the best setups do.\n\nSee the adjacent quests for more information on computer racks and valid rack components."
```
Comment (chochem) [inline config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumComputer-AAAAAAAAAAAAAAAAAAAH1w==.json:38]: not suggesting a basic sample setup without need for parameters to get people started is a pretty bad change imo.

File: config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumMechanics-AAAAAAAAAAAAAAAAAAAKiw==.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
...ile it is hot (\u003e2000) will similarly void the components inside. The QC will explode if the heat exceeds 10,000 but no component has a heat limit greater than that, so don\u0027t worry too much about ruining your base.\n\n[note]Heat α HeatConstant * overclock * overvolt^2[/note]\n\nThe power consumption of a rack is 1A ZPM regardless of the components inside. This is in addition to the 1A ZPM of the QC itself. However, these EU/t values are directly proportional to overclock and overvolt which should increase as you unlock better circuits. Be sure to turn on the QC before the research station and only when you need it.\n\n[note]EU/t \u003d 131,072 * overclock * overvolt * (racks + 1)[/note]\n\nTo get you started, craft 4x APU (Tier 3) and 4x Cooling Cores. Place two of each in the two racks. Set the overclock and overvolt values to 1.44 and 1.03, respectively. This will generate 1,995 computation/s at a cost of 583,217 EU/t or just over 1A of UV which is more than enough for researching the Wetware Mainframe. If the APUs are too bothersome to craft, you can use Crystalprocessor Mainframes instead but you'll need to find some appropriate overclock and overvolt values yourself."
```
Comment (chochem) [inline config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumMechanics-AAAAAAAAAAAAAAAAAAAKiw==.json:32]: 2-24

File: config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/TheNumbersMasonW-zzIqlbfdQl6Rn8cVlbiQsA==.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
```
Comment (chochem) [inline config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/TheNumbersMasonW-zzIqlbfdQl6Rn8cVlbiQsA==.json:33]: I dont think this quest is a good idea

**Comment by DylanTaylor1** [comment]
Alrighty no worries, I deleted the quest with all the rack component values and adjusted the example setup.

File: config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumComputer-AAAAAAAAAAAAAAAAAAAH1w==.json (hunk 1 above)
Comment (chochem) [reply config/betterquesting/DefaultQuests/Quests/Tier8UV-AAAAAAAAAAAAAAAAAAAALQ==/QuantumComputer-AAAAAAAAAAAAAAAAAAAH1w==.json]: yea having it in the other quest is fine.

**Comment by chochem** [comment]
I am also happy to hear other opinions on how to best represent that data:
- custom nei page (similar to how fuel values are shown)
- wiki
- tooltip? maybe not great as its just one small use from the perspective of a circuit.
- a table in the QB?

but yea for me at least a table in the QB does not seem like a great option. QB can/should of course have a link to the wiki if the data is there.

**Review by Dream-Master — APPROVED** [review]

**Pull request merged**